File: BlackVision/Applications/ProjectManagerPrototype/AssetCategory.py
from AssetAccessor import AssetAccessor
from AssetDesc import AssetDesc

class AssetCategory:
    def __init__(self, id, assetAccessor):
        assert isinstance(assetAccessor, AssetAccessor)
        self.id = id
        self.accessor = assetAccessor

    def getId(self):
        return self.id

    def appendAsset(self, path, assetDesc):
        assert isinstance(assetDesc, AssetDesc)
        self.accessor.appendAsset(path, assetDesc)

    def getAssetDesc(self, path):
        assert isinstance(path, str)
        return self.accessor.getLoadableAssetDesc(path)

    def removeAsset(self, path):
        assert isinstance(path, str)
        return self.accessor.removeAsset(path)

    def exportAsset(self, expAssetFilePath, path):
        assert isinstance(path, str)
        return self.accessor.exportAsset(expAssetFilePath, path)

    def importAsset(self, impAssetFilePath, toPath):
        assert isinstance(toPath, str)
        return self.accessor.importAsset(impAssetFilePath, toPath)

    def exportAll(self, expAssetFilePath):
        return self.accessor.exportAll(expAssetFilePath)

    def listAssets(self):
        return self.accessor.listAll()

File: BlackVision/Applications/BlackVision/Source/ProjectManager/AssetCategory.cpp
#include "AssetCategory.h"
#include "AssetAccessor.h"

namespace bv
{

// ********************************
//
AssetCategory::AssetCategory( AssetAccessor * aa )
	: m_assetAccessor( aa )
{}

// ********************************
//
AssetCategory::~AssetCategory()
{}

// ********************************
//
const std::string &		AssetCategory::GetId() const
{
	return m_id;
}

// ********************************
//
void					AssetCategory::AddAsset	( const NormalizedPath & path, const AssetDesc & assetDesc )
{
	m_assetAccessor->AddAsset( path, assetDesc );
}

// ********************************
//
const AssetDesc *		AssetCategory::GetAssetDesc	( const NormalizedPath & path ) const
{
	return m_assetAccessor->GetAssetDesc( path );
}

// ********************************
//
void					AssetCategory::RemoveAsset	( const NormalizedPath & path )
{
	m_assetAccessor->RemoveAsset( path );
}

// ********************************
//
void					AssetCategory::ExportAsset	( const NormalizedPath & expAssetFilePath, const NormalizedPath & path ) const
{
	m_assetAccessor->ExportAsset( expAssetFilePath, path );
}

// ********************************
//
void					AssetCategory::ImportAsset	( const NormalizedPath & impAssetFilePath, const NormalizedPath & toPath )
{
	m_assetAccessor->ImportAsset( impAssetFilePath, toPath );
}

// ********************************
//
void					AssetCategory::ExportAll	( const NormalizedPath & expAssetFilePath ) const
{
	m_assetAccessor->ExportAll( expAssetFilePath );
}

// ********************************
//
void					AssetCategory::ListAssets	( const NormalizedPath & projectName ) const
{
	m_assetAccessor->ListAll( projectName );
}

} // bv